### PR TITLE
Add a demo for multiple configs provided to a Vector instance.

### PR DESCRIPTION
**Pull request by silversupreme**
We've had customers ask how to do this; this should provide a simple example to give them for how to set it up, and demonstrate part of the value for Observability Pipeline's remote config management.

### CODE REVIEW

File: multi-config/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+from a directory and adds that to the supported `extraConfigDirSources` for the Vector pod.
+
+To add configs to Vector in this setup, all you must do is add them to the `./vector` directory: both the locally-running
+Vector as well as the one installed by the Helm chart are configured to utilize all files there. Only `.yaml` files are
+supported for now.
+
+A simple `Makefile` is supplied which demonstrates the proper commands to test this setup both locally and through Helm.
+Teardown commands are also provided there for when testing is complete.
+
+The Helm chart was tested in `minikube` running on an EC2 instance.
```
Comment (blt) [inline multi-config/README.md:23]: Might list which version of minikube, for completeness. Same story for which OS on the EC2 instance.

**Review by blt — APPROVED** [review]
Very minor comment, otherwise +1.

File: multi-config/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v2
+name: multi-config-vector
+version: 1.0.0
+description: A demonstration of multiple ConfigMaps being merged into a single Vector config.
+dependencies:
+  - name: vector-aggregator
+    version: ~0.19.1
```
Comment (spencergilbert) [inline multi-config/Chart.yaml:7]: As noted offline this should be changed to the new chart: https://github.com/vectordotdev/helm-charts/blob/develop/charts/vector/README.md

```suggestion
  - name: vector
    version: ~0.2.3
```

File: multi-config/templates/configs.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
```
Comment (spencergilbert) [inline multi-config/templates/configs.yaml:1]: There is a [size limit](https://kubernetes.io/docs/concepts/configuration/configmap/#:~:text=A%20ConfigMap%20is%20not%20designed,separate%20database%20or%20file%20service.) here that users should be aware of, definitely can hit it with a couple large configs.

You could alternatively create a ConfigMap for each of the files provided in the `vector/` dir.

File: multi-config/values.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+vector-aggregator:
+  extraConfigDirSources:
+    - configMap:
+        name: configs
```
Comment (spencergilbert) [inline multi-config/values.yaml:4]: For the new chart this would be...

```suggestion
vector:
  existingConfigMaps:
    - configs
```

**Review by spencergilbert — CHANGES_REQUESTED** [review]
Additionally the new chart will make you decide on the data_dir and api config so another config file, perhaps `vector/global.yaml`:

```yaml
data_dir: /vector-data-dir
api:
  enabled: true
  address: 127.0.0.1:8686
  playground: false
```

I think this should work

File: multi-config/templates/configs.yaml (hunk 3 above)
Comment (silversupreme) [reply multi-config/templates/configs.yaml:1]: This one's kinda tricky honestly.

Though the `ConfigMap` templates definitely support the Go templating that would be necessary for that, the `values.yaml` that adds them to Vector does not. [It's a major issue that Helm has](https://github.com/helm/helm/issues/2492), and one that I call out in the `README`.

I'll a note about this to the README, even more evidence that Observability Pipelines has much better tooling for handling this. 😄

Comment (spencergilbert) [reply multi-config/templates/configs.yaml:1]: True - though you could manually add them into to vector's `values.yaml` or do some awful shelling and seding to generate it before running helm commands

File: multi-config/README.md [hunk 5]
```diff
@@ -1,17 +1,19 @@
...
+management by multiple business units.
 
 ## Constraints
 * **This solution works inside of a vanilla Kubernetes cluster.** This is intended as a simple solution that any
   customer can easily hit the ground running with.
 * **Helm does not support templating in `values.yaml`.** [The method we use to provide extra config files to Vector does not support templated names](https://github.com/helm/helm/issues/2492),
   which makes it difficult to make this a truly generic solution for your needs. In particular, this makes it
-  impossible to properly scope the ConfigMap names with `{{ .Release.Name }}` as you might wish.
+  impossible to properly scope the ConfigMap names with `{{ .Release.Name }}` as you might wish, or to mount each config
+  file as a separate ConfigMap to avoid [file size limits](https://kubernetes.io/docs/concepts/configuration/configmap/#:~:text=A%20ConfigMap%20is%20not%20designed,separate%20database%20or%20file%20service.).
```
Comment (spencergilbert) [inline multi-config/README.md:12]: Realizing that this link may only work on chrome 🤔

**Review by spencergilbert — APPROVED** [review]
Did I actually get all of the suggestions right (aside from the version for the vector chart)? I didn't even double check 😆

**Pull request merged**